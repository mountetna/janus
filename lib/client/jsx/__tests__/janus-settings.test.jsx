import React from 'react';
import {Provider} from 'react-redux';
import { act, create } from 'react-test-renderer';
import {mockStore} from '../test-helpers';
import {stubUrl} from 'etna-js/spec/helpers';
import JanusSettings from '../janus-settings';

describe('JanusSettings', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      user: {
	email: "janus@two-faces.org",
	name: "Janus Bifrons",
	permissions: { }
      }
    });
  });

  it('renders', async () => {
    const initialStubs = [
      stubUrl({
        verb: 'get',
        path: '/user',
        status: 200,
        response: {
          user: {
            email: "janus@two-faces.org",                                  
            name: "Janus Bifrons",
            public_key: '74:68:69:73:69:73:6e:6f:74:72:61:6e:64:6f:6d:21'
          }
        }
      })
    ];

    let component = create(
      <Provider store={store}>
        <JanusSettings/>
      </Provider>
    );

    await act( async () => {
        await new Promise((resolve) => setTimeout(resolve, 15));
    });

    expect( component.toJSON() ).toMatchSnapshot()
  });
});
