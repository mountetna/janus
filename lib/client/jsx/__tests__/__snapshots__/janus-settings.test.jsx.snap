// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JanusSettings renders 1`] = `
<div
  id="janus-settings"
>
  <div
    id="keys-group"
  >
    <div
      className="title"
    >
      Your Keys
    </div>
    <div
      className="item"
    >
      <i
        className="fa fa-key"
      />
       
      74:68:69:73:69:73:6e:6f:74:72:61:6e:64:6f:6d:21
    </div>
    <div
      className="item"
    >
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-text-1"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Paste 2048+ bit RSA key in PEM format"
            required={false}
            type="text"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Upload Key
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    id="task-token-group"
  >
    <div
      className="title"
    >
      Task Tokens
    </div>
    <div
      className="item"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-text-1"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          tabIndex={-1}
          value={null}
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={null}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Copy Task Token
        </span>
      </button>
    </div>
  </div>
</div>
`;
